napari/layers/points/points.py

```diff
@@ -2033,8 +2033,8 @@ def _paste_data(self):
             self._selected_view = list(
                 range(npoints, npoints + len(self._clipboard['data']))
             )
-            self._selected_data = set(
-                range(totpoints, totpoints + len(self._clipboard['data']))
+            self._selected_data = Selection(
+                set(range(totpoints, totpoints + len(self._clipboard['data'])))
```

Comment:
actually, I think this should update the selectio n in place, to avoid issues with connected events losing the source... That's probably broken in main ATM.

Reply:
Fixed.